message_adapter/cumulus_message.py

```diff
@@ -0,0 +1,263 @@
+    *   1. Single curly-braces within a string ("some{$.path}value"). The JSONPaths
+    *      are replaced by the first value they match, coerced to string
+    *   2. A string surrounded by double curly-braces ("{{$.path}}").  The function
+    *      returns the first object matched by the JSONPath
+    *   3. A string surrounded by curly and square braces ("{[$.path]}"). The function
+    *      returns an array of all object matching the JSONPath
+    *
+    * It's likely we'll need some sort of bracket-escaping at some point down the line
+    *
+    * @param {*} event The Cumulus message
+    * @param {*} json_path_string A string containing a JSONPath template to resolve
+    * @returns {*} The resolved object
+    """
+    value_regex = r"^{[^\[\]].*}$"
+    array_regex = r"^{\[.*\]}$"
+    template_regex = '{[^}]+}'
+
+    if re.search(value_regex, json_path_string):
+        match_data = parse(json_path_string.lstrip('{').rstrip('}')).find(event)
+        return match_data[0].value if match_data else None
+
+    if re.search(array_regex, json_path_string):
+        parsed_json_path = json_path_string.lstrip('{').rstrip('}').lstrip('[').rstrip(']')
+        match_data = parse(parsed_json_path).find(event)
+        return [item.value for item in match_data] if match_data else []
+
+    if re.search(template_regex, json_path_string):
+        matches = re.findall(template_regex, json_path_string)
+        for match in matches:
+            match_data = parse(match.lstrip('{').rstrip('}')).find(event)
+            if match_data:
+                json_path_string = json_path_string.replace(match, match_data[0].value)
+        return json_path_string
+
+    return json_path_string
+
+
+def resolve_input(event, config):
+    """
+    * Given a Cumulus message and its config, returns the input object to send to the
+    * task, as defined under config.cumulus_message
+    * @param {*} event The Cumulus message
+    * @param {*} config The config object
+    * @returns {*} The object to place on the input key of the task's event
+    """
+    if ('cumulus_message' in config and 'input' in config['cumulus_message']):
+        input_path = config['cumulus_message']['input']
+        return resolve_path_str(event, input_path)
+    return event.get('payload')
+
+
+def resolve_config_templates(event, config):
+    """
+    * Given a config object containing possible JSONPath-templated values, resolves
+    * all the values in the object using JSONPaths into the provided event.
+    *
+    * @param {*} event The event that paths resolve against
+    * @param {*} config A config object, containing paths
+    * @returns {*} A config object with all JSONPaths resolved
+    """
+    task_config = config.copy()
+    if 'cumulus_message' in task_config:
+        del task_config['cumulus_message']
+    return _resolve_config_object(event, task_config)
+
+
+def store_remote_response(incoming_event, max_size, config_keys):
```

Comment:
maybe rename `max_size` to `default_max_size` because down the line it can come from either the `replace_config` or this value so the name is slightly overloaded.

Reply:
Agreed 3f429c6